### PR TITLE
[TACHYON-142] support viewing log files from the web UI

### PR DESCRIPTION
**Pull request by gpang**
This adds links to both the master web UI and the worker web UI to browse the logs/ directory. It reuses much of the the existing browse files interface to stay consistent. Therefore, it can also seek to different offsets of the log files as well as download the entire file.

### CODE REVIEW

**Comment by gpang** [comment]
Here are some screenshots. Don't mind all the fake log files...

screen shot 2015-04-11 at 4 01 59 pm
screen shot 2015-04-11 at 4 02 17 pm

File: core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+   * @param path The path of the file to display
+   * @param request The HttpServletRequest object
+   * @param offset Where the file starts to display.
+   * @throws IOException
+   */
+  private void displayLocalFile(Path path, HttpServletRequest request, long offset)
+      throws IOException {
+    String fileData = null;
+    InputStream is = Files.newInputStream(path);
+    long fileSize = Files.size(path);
```
Comment (calvinjia) [inline core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java]: Could this be done in the try block?

File: core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    request.setAttribute("baseUrl", "./browseLogs");
+    request.setAttribute("currentPath", "");
+
+    String baseDir = mTachyonConf.get(Constants.TACHYON_HOME, Constants.DEFAULT_HOME);
+
+    String requestFile = request.getParameter("path");
+
+    if (requestFile == null || requestFile.isEmpty()) {
+      // List all log files in the log/ directory.
+      DirectoryStream<Path> directories =
```
Comment (calvinjia) [inline core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java]: Would it be more accurate to call these `logFiles`?

File: core/src/main/java/tachyon/web/WebInterfaceDownloadLocalServlet.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    String requestPath = request.getParameter("path");
+    if (requestPath == null || requestPath.isEmpty()) {
+      requestPath = TachyonURI.SEPARATOR;
+    }
+
+    // Download a file from the local filesystem.
+    TachyonConf tachyonConf = new TachyonConf();
```
Comment (calvinjia) [inline core/src/main/java/tachyon/web/WebInterfaceDownloadLocalServlet.java]: Could this be done in the constructor?

**Comment by calvinjia** [comment]
Do you think it makes sense to split up the view one log file in detail into a different servlet than the view all logs in directory format? Also, what do you think about keeping the header and footer outside of the body page?

File: core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java (hunk 1 above)
Comment (gpang) [inline core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java]: good catch. done.

File: core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java (hunk 2 above)
Comment (gpang) [inline core/src/main/java/tachyon/web/WebInterfaceBrowseLogsServlet.java]: done

File: core/src/main/java/tachyon/web/WebInterfaceDownloadLocalServlet.java (hunk 3 above)
Comment (gpang) [inline core/src/main/java/tachyon/web/WebInterfaceDownloadLocalServlet.java]: done

**Comment by gpang** [comment]
Right now, the browsing and viewfiles was in the same servlet, so I just copied that convention. I think we can change that in a separate PR to separate them both.

As for the jsp body/header/footer splitting, I am not exactly sure what you are suggesting. Are you saying that you want something to end up like:

<include header>
<include body>
<include footer>

If so, then it probably won't be so clean given the jsp's now. If not, then I am not sure what you want with the jsp's.

**Comment by calvinjia** [comment]
Yeah, those changes are not in the scope of this PR.
+1 LGTM

**Comment by haoyuan** [comment]
@shaoshanliu

**Pull request merged**